tasks/update_go.py

```diff
@@ -120,15 +126,106 @@ def _handle_file(path: str, patterns: Dict[re.Pattern, str], expected_match: int
+def _add_wait_for_tests(branch: str):
+    import re
+    trigger = """project: DataDog/datadog-agent"""
+    # load bearing yaml whitespace :sigh:
+    with_branch = f"""{trigger}
+    branch: {branch}"""
+    with open(".gitlab-ci.yml", "r") as f:
+        contents = f.read()
+    contents = re.sub(trigger, with_branch, contents, count=1)
+    with open(".gitlab-ci.yml", "w") as f:
+        f.write(contents)
+
+# TODO: not sure how to run this before merging
+def _remove_wait_for_tests(branch: str):
+    import re
+    with_branch = f"""project: DataDog/datadog-agent
+    branch: .*"""
+    with open(".gitlab-ci.yml", "r") as f:
+        contents = f.read()
+    contents = re.sub(with_branch, "project: DataDog/datadog-agent", contents, count=1)
+    with open(".gitlab-ci.yml", "w") as f:
+        f.write(contents)
+
+def _update_and_create_pr(gover):
+    # TODO: check that appgate is turned on so that gitlab.ddbuild.io is reachable
+    repo = "datadog-agent-buildimages"
+    branch = f"update-go-{gover}"
+    with dd_repo_temp_cwd(repo):
+        if local_uncommited_changes_exist(repo):
+            raise exceptions.Exit("Uncommited changes exist in datadog-agent-buildimages repo. Exiting.")
+        checkout_latest_main(repo)
+        _add_wait_for_tests(branch)
+        _update_images(gover)
+        if local_uncommited_changes_exist(repo):
+            create_branch_and_push_changes(
+                repo,
+                branch,
+                f"Update Go version to {gover} for datadog agent buildimages",
+            )
+        else:
+            raise exceptions.Exit(f"Updating buildimages to {gover} changed nothing. Exiting.")
+
+        # gitlab job takes a bit to start up
+        # GIT_COMMIT_SHORT_SHA used in gitlab is first 8 characters of commit SHA
+        commit_sha = subprocess.check_output(["git", "rev-parse", "@"]).strip().decode('utf-8')
+        print(commit_sha)
+        tries = 10
+        build_ids = []
+        for i in range(tries):
+            status = requests.get(f"https://api.github.com/repos/DataDog/datadog-agent-buildimages/commits/{commit_sha}/status", headers={"Cache-Control": "no-cache"}).text
+            print(status)
+            matches = re.findall("\"https://gitlab.ddbuild.io/datadog/datadog-agent-buildimages/builds/(\d+)\"", status)
+            if len(matches) < 1:
+                if i < tries - 1:
+                    time.sleep(15)
+                    continue
+                else:
+                    raise exceptions.Exit("Exhausted retries waiting for gitlab job to start.")
+            else:
+                build_ids.extend(matches)
+                print(f"Gitlab job started with build_ids: {build_ids}")
+                break
+
+        # XXX: is it OK to require the user create a gitlab access token?
+        res = requests.get(f"https://gitlab.ddbuild.io/api/v4/projects/291/jobs/{build_ids[0]}", headers={"PRIVATE-TOKEN": os.environ["GITLAB_ACCESS_TOKEN"]})
```

Comment:
it would be very nice to determine the gitlab pipeline id without needing an access_token but I just [used the gitlab api](https://docs.gitlab.com/ee/api/jobs.html#get-a-single-job)